### PR TITLE
Removing NULL API response attribute from self-registration API

### PR DESCRIPTION
**Pull request by somindatommy**
### Proposed changes in this pull request

Resolves https://github.com/wso2/product-is/issues/7262

Changed the API response when the notification channel is either `SMS` or `EMAIL`.
```
{
"code": "USR-02001",
"message": "Successful user self registration. Pending account verification",
"notificationChannel": "EMAIL",
}
```

### CODE REVIEW

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java [hunk 1]
```diff
@@ -1,95 +1,86 @@
...
-  }
-  public void setMessage(String message) {
-    this.message = message;
-  }
+    /**
+    * Status code of the operation.
+    **/
+    @ApiModelProperty(value = "Status code of the operation.")
+    @JsonProperty("code")
+    public String getCode() {
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: New line after method signature

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java [hunk 2]
```diff
@@ -65,6 +67,33 @@ public void testConflictException() {
         assertEquals(conflictException.getResponse().getStatus(), 409);
     }
 
+    @Test
+    public void testHandleConflictException(){
 
-}
+        ConflictException conflictException = new ConflictException();
+        assertEquals(conflictException.getResponse().getStatus(),409);
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: space after comma

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java (hunk 1 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: These are auto-generated classes. It's better if we did not change them.

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java [hunk 3]
```diff
@@ -65,6 +67,33 @@ public void testConflictException() {
...
+        String errorCode = "UNR-10002";
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: Aren't there any constants for error codes

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java [hunk 4]
```diff
@@ -1,95 +1,86 @@
...
-    sb.append("  code: ").append(code).append("\n");
-    sb.append("  message: ").append(message).append("\n");
-    sb.append("  notificationChannel: ").append(notificationChannel).append("\n");
-    sb.append("  confirmationCode: ").append(confirmationCode).append("\n");
-    sb.append("}\n");
-    return sb.toString();
-  }
+        StringBuilder sb = new StringBuilder();
+        sb.append("class SuccessfulUserCreationDTO {\n");
+
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: shall we remove these new lines

File: components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java [hunk 5]
```diff
@@ -112,15 +117,24 @@ private Response buildSuccessfulAPIResponse(NotificationResponseBean notificatio
 
         // Check whether detailed api responses are enabled.
         if (isDetailedResponseBodyEnabled()) {
-            SuccessfulUserCreationDTO successfulUserCreationDTO = buildSuccessResponse(notificationResponseBean);
+            String notificationChannel = notificationResponseBean.getNotificationChannel();
+            if (StringUtils.isNotEmpty(notificationChannel) && notificationChannel
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java]: Let's use NotificationChannels.EXTERNAL_CHANNEL.getChannelType().equals(notificationChannel). Then we can get rid of the redundant StringUtils.isNotEmpty(notificationChannel) check

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java [hunk 6]
```diff
@@ -1,95 +1,86 @@
...
-    sb.append("}\n");
-    return sb.toString();
-  }
+        StringBuilder sb = new StringBuilder();
+        sb.append("class SuccessfulUserCreationDTO {\n");
+        
+        sb.append("    code: ").append(code).append("\n");
+        sb.append("    message: ").append(message).append("\n");
+        sb.append("    notificationChannel: ").append(notificationChannel).append("\n");
+
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: Shall we remove this new line, unwanted break

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java [hunk 7]
```diff
@@ -1,95 +1,86 @@
...
+    /**
+    * Account confirmation notification sent channel.
+    **/
+    @ApiModelProperty(value = "Account confirmation notification sent channel.")
+    @JsonProperty("notificationChannel")
+    public String getNotificationChannel() {
+        return notificationChannel;
+    }
+    public void setNotificationChannel(String notificationChannel) {
+        this.notificationChannel = notificationChannel;
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: Need new line before method signature, please check the other methods too

File: components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java [hunk 8]
```diff
@@ -112,15 +117,24 @@ private Response buildSuccessfulAPIResponse(NotificationResponseBean notificatio
...
+                    buildSuccessResponseForInternalChannels(notificationResponseBean);
             return Response.status(Response.Status.CREATED).entity(successfulUserCreationDTO).build();
         } else {
             if (notificationResponseBean != null) {
                 String notificationChannel = notificationResponseBean.getNotificationChannel();
                 // If the notifications are required in the form of legacy response, and notifications are externally
                 // managed, the recoveryId should be in the response as text.
                 if (StringUtils.isNotEmpty(notificationChannel) && notificationChannel
-                        .equals(Constants.EXTERNAL_NOTIFICATION_CHANNEL)) {
+                        .equals(NotificationChannels.EXTERNAL_CHANNEL.getChannelType())) {
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/MeApiServiceImpl.java]: we can get rid of StringUtils.isNotEmpty(notificationChannel)  by interchaning the variable with the constant as suggested above.

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java [hunk 9]
```diff
@@ -1,95 +1,86 @@
...
-  }
+    /**
+    * Descriptive message regarding the operation.
+    **/
+    @ApiModelProperty(value = "Descriptive message regarding the operation.")
+    @JsonProperty("message")
+    public String getMessage() {
+        return message;
+    }
+    public void setMessage(String message) {
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: Shall we add method description comment to all public methods at least and private methods if needed

Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: These are auto-generated classes. It's better if we did not change them.

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java (hunk 7 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: These are auto-generated classes. It's better if we did not change them.

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java [hunk 10]
```diff
@@ -65,6 +67,33 @@ public void testConflictException() {
...
+        String errorMessage = "Test Error";
+        try {
+            Utils.handleConflict(errorMessage, errorCode);
+        } catch (ConflictException exception) {
+            assertEquals(exception.getResponse().getStatus(),409);
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: formatting issue

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java (hunk 6 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: These are auto-generated classes. It's better if we did not change them.

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java [hunk 11]
```diff
@@ -65,6 +67,33 @@ public void testConflictException() {
...
+            assertEquals(exception.getMessage(),errorMessage);
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: formatting issue

File: components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java (hunk 4 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/gen/java/org/wso2/carbon/identity/user/endpoint/dto/SuccessfulUserCreationDTO.java]: These are auto-generated classes. It's better if we did not change them.

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java [hunk 12]
```diff
@@ -65,6 +67,33 @@ public void testConflictException() {
...
+        }
+    }
 
+    @Test
+    public void testHandleNotAcceptableException(){
+
+        NotAcceptableException notAcceptableException = new NotAcceptableException();
+        assertEquals(notAcceptableException.getResponse().getStatus(),406);
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: formatting issue

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java [hunk 13]
```diff
@@ -65,6 +67,33 @@ public void testConflictException() {
...
+        String errorCode = "UNR-10002";
+        String errorMessage = "Test Error";
+        try {
+            Utils.handleNotAcceptable(errorMessage, errorCode);
+        } catch (NotAcceptableException exception) {
+            assertEquals(exception.getResponse().getStatus(),406);
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: formatting issues in line 95 and 96

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java (hunk 3 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: These are just test codes. Will change the value to reflect it.

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java (hunk 13 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: Fixed: f929c4ac1bd444a80c901a7c4281882c7728403a

File: components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java (hunk 12 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/test/java/org.wso2.carbon.identity.user.endpoint.Util/HandleExceptionsTest.java]: Fixed: f929c4ac1bd444a80c901a7c4281882c7728403a

**Pull request merged**